File: backend/src/main/java/com/example/app/entity/Note.java
package com.example.app.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "notes", indexes = {
    @Index(name = "idx_notes_user_id", columnList = "user_id"),
    @Index(name = "idx_notes_user_pinned", columnList = "user_id, pinned"),
    @Index(name = "idx_notes_user_updated", columnList = "user_id, updated_at DESC")
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Note {

    @Id
    @Column(length = 36)
    private String id;

    @Column(name = "user_id", nullable = false, length = 36)
    private String userId;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(length = 50)
    @Builder.Default
    private String category = "默认";

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "JSON")
    @Builder.Default
    private List<String> tags = List.of();

    @Builder.Default
    private Boolean pinned = false;

    @Column(name = "memory_id", length = 36)
    private String memoryId;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: KchatMac/KchatMac/Data/NoteTodo/Note.swift
import SwiftData
import Foundation

@Model
final class Note {
    @Attribute(.unique) var id: String
    var userId: String
    var title: String
    var content: String
    var category: String
    var tags: [String]
    var pinned: Bool
    var createdAt: Date
    var updatedAt: Date
    
    init(id: String, title: String, content: String) {
        self.id = id
        self.userId = "default"
        self.title = title
        self.content = content
        self.category = "default"
        self.tags = []
        self.pinned = false
        self.createdAt = Date()
        self.updatedAt = Date()
    }
    
    func update(title: String? = nil, content: String? = nil, category: String? = nil, 
                tags: [String]? = nil, pinned: Bool? = nil) {
        if let title { self.title = title }
        if let content { self.content = content }
        if let category { self.category = category }
        if let tags { self.tags = tags }
        if let pinned { self.pinned = pinned }
        self.updatedAt = Date()
    }
}